From: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
Date: Fri, 30 Sep 2022 20:41:26 +0200
Subject: tests: Disable bits that require the threads.

A few tests require thread support to be compiled. This does not work if
the project was configured with --disable-threads.

Disable/ Skip tests which require thead support to be comppiled as part
of the build.

Signed-off-by: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
---
 tests/test_check.c    | 5 ++++-
 tests/test_hardware.c | 5 ++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/tests/test_check.c b/tests/test_check.c
index fa1326d..ae385be 100644
--- a/tests/test_check.c
+++ b/tests/test_check.c
@@ -232,6 +232,7 @@ test_lzma_get_check_st(void)
 }
 
 
+#ifdef MYTHREAD_ENABLED
 static void
 test_lzma_get_check_mt(void)
 {
@@ -312,7 +313,7 @@ test_lzma_get_check_mt(void)
 
 	lzma_end(&strm);
 }
-
+#endif
 
 extern int
 main(int argc, char **argv)
@@ -346,7 +347,9 @@ main(int argc, char **argv)
 	tuktest_run(test_lzma_supported_checks);
 	tuktest_run(test_lzma_check_size);
 	tuktest_run(test_lzma_get_check_st);
+#ifdef MYTHREAD_ENABLED
 	tuktest_run(test_lzma_get_check_mt);
+#endif
 
 	return tuktest_end();
 }
diff --git a/tests/test_hardware.c b/tests/test_hardware.c
index 34d0789..aa0d290 100644
--- a/tests/test_hardware.c
+++ b/tests/test_hardware.c
@@ -27,19 +27,22 @@ test_lzma_physmem(void)
 }
 
 
+#ifdef MYTHREAD_ENABLED
 static void
 test_lzma_cputhreads(void)
 {
 	if (lzma_cputhreads() == 0)
 		assert_skip("Could not determine cpu core count");
 }
-
+#endif
 
 extern int
 main(int argc, char **argv)
 {
 	tuktest_start(argc, argv);
 	tuktest_run(test_lzma_physmem);
+#ifdef MYTHREAD_ENABLED
 	tuktest_run(test_lzma_cputhreads);
+#endif
 	return tuktest_end();
 }
